### ISSUE TITLE
S3InputFormat support buckets with "." in their name

### ISSUE THREAD

**Issue by ehochmuth** [locationtech/geotrellis#2132]
The current regular expression in S3InputFormat that parses the bucket and key/prefix only allows bucket names with "-". Extends the S3InputFormat regular expression to support both "-" and "." in bucket names.

https://github.com/locationtech/geotrellis/blob/master/s3/src/main/scala/geotrellis/spark/io/s3/S3InputFormat.scala#L200

https://github.com/locationtech/geotrellis/pull/2133

**Comment by notthatbreezy** [comment]
Does it make sense to lean on a standard library class to limit other edge cases like this?

For instance, `URI` seems to satisfy the functionality here:

```scala
scala> import java.net.URI
import java.net.URI

scala> val s3URI = new URI("s3://username:password@bucket.with.periods/path/to/objects")
s3URI: java.net.URI = s3://username:password@bucket.with.periods/path/to/objects

scala> s3URI.getHost
res10: String = bucket.with.periods

scala> s3URI.getPath
res11: String = /path/to/objects

scala> s3URI.getScheme
res12: String = s3

scala> s3URI.getUserInfo
res13: String = username:password
```

**Comment by ehochmuth** [comment]
Or make the username and password part of the backend-profile and then just let amazon do the work:  com.amazonaws.services.s3.AmazonS3URI

**Comment by ehochmuth** [comment]
Moving over to Pull Request proposal: #2133

**Comment by lossyrob** [comment]
Thanks @ehochmuth. We usually keep an issue open, and the PR can say "Fixes #XXXX" in the description, which will automatically close the issue when the PR gets merged. Reopening and setting that in the description.

**Comment by lossyrob** [comment]
Fixed by #2133, not sure why it didn't close automatically.
